### DISCUSSION TITLE
`no_superfluous_phpdoc_tags` behavior question

### DISCUSSION THREAD

**Discussion by mbolli** [PHP-CS-Fixer/PHP-CS-Fixer#8110] [Q&A] [answered]
Is it correct that `no_superfluous_phpdoc_tags` rule does nothing in this case:

```php
/**
 * @return ?float
 */
public function getFloatValue(): ?float { ... }
```

But does remove the `@return` in this case?

```php
/**
 * @return null|float
 */
public function getFloatValue(): ?float { ... }
```

**Comment by julienfalque** [comment]
As far as I remember, PHP CS Fixer does not recognize `?foo` as equivalent to `foo|null` in PHPDoc.

**Reply by keradus** [reply]
the question is if it should, at least for this rule

**Comment by keradus** [comment, marked as answer]
-=> https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8125

**Reply by mbolli** [reply]
Great, thanks!
